src/objects.js

```diff
@@ -13491,3 +13491,497 @@ ReplayControls.prototype.getInverseEvent = function(event) {
         }
     }
 };
+
+////// READING MUSICXML //////
+
+/**
+ * @description - Creates and instance of an object representing a note in a musical work.
+ * @param {Element} src - The contents of a <note> tag in a .musicxml file. 
+ */
+function Note(src) {
+    this.src = src;
+
+    this.initPitch = function () {
+        if (this.src.getElementsByTagName("rest").length > 0 || 
+        this.src.getElementsByTagName("step").length == 0 ||
+        this.src.getElementsByTagName("octave").length == 0) {
+            return "Rest";
+        }
+        const step = this.src.getElementsByTagName("step")[0].childNodes[0].nodeValue;
+        const octave = this.src.getElementsByTagName("octave")[0].childNodes[0].nodeValue;
+        return step + octave;
+    }
+
+    this.initType = function () {
+        const element = this.src.getElementsByTagName("type");
+        if (element.length == 0) {
+            return "Whole";
+        }
+        const temp = element[0].childNodes[0].nodeValue;
+        return temp.charAt(0).toUpperCase() + temp.slice(1);
+    }
+
+    this.initDotted = function () {
+        const element = this.src.getElementsByTagName("dot");
+        return element.length != 0;
+    }
+
+    this.pitch = this.initPitch();
+    this.type = this.initType();
+    this.dotted = this.initDotted();
+}
+
+/**
+ * @description - Creates and instance of an object representing a part in a musical work.
+ * @param {Element} src - The contents of a <part> tag in a .musicxml file. 
+ * @param {String} name - The name of the part.
+ */
+function Part(src, name="untitled") {
+    this.src = src;
+    this.notes = [];
+    this.name = name;
+    const xmlMeasures = this.src.getElementsByTagName("measure");
+    for (let i = 0; i < xmlMeasures.length; ++i) {
+        const xmlNotes = xmlMeasures[i].getElementsByTagName("note");
+        for (let j = 0; j < xmlNotes.length; ++j) this.notes.push(new Note(xmlNotes[j]));
+    }
+}
+
+/**
+ * @description - Creates a MusicReader object that parses MusicXML files.
+ * @constructor
+ * @param {String} src - The contents of a .musicxml file.
+ * @param {String} name - The name of the MusicReader.
+ */
+function MusicReader(src, name = "untitled") {
+    this.src = src;
+    this.name = name;
+    this.parts = [];
+    const parser = new DOMParser();
+    const xmlDoc = parser.parseFromString(this.src, "text/xml");
+    const parts = xmlDoc.getElementsByTagName("part");
+    const scoreParts = xmlDoc.getElementsByTagName("part-name");
+    for (let i = 0; i < parts.length; ++i) {
+        this.parts.push(new Part(parts[i], scoreParts[i].textContent));
+    }
+}
+
+/**
+ * @description - Creates a MusicWriter object with the data from a MusicReader.
+ * @constructor
+ * @param {MusicReader} musicReader - The contents of a MusicReader.
+ */
+function MusicWriter(musicReader) {
+    const DEFAULT_TEMP0 = 60;
+    this.mMusicReader = musicReader;
+    this.myRoot = new Document();
```

Comment:
When possible, I would recommend that you avoid adding unnecessary state to the object. Instead, I would add the current state of the output to the methods that add to it. This will make writing automated unit tests (in the test/ directory) easier.